LinqToRest.Test/ResourceRetrieverTest.cs

```diff
@@ -0,0 +1,193 @@
+
+            return queryBinderFactory;
+        }
+
+        #endregion
+
+        #region Data
+
+        private static Uri RootUri => new Uri("http://www.example.com/api/v1/", UriKind.Absolute);
+
+        private static string EntityWithQueryableMemberRequestUri =>
+            $"{RootUri}entitywithqueryablemembers";
+
+        private static string EntityWithQueryableMemberJson => @"
+[
+    {
+        ""uniqueIdentifier"": ""Test1"",
+        ""name"": ""Test1""
+    },
+    {
+        ""uniqueIdentifier"": ""Test2"",
+        ""name"": ""Test2""
+    }
+]
+";
+
+        private static QueryResult<object> EntityWithQueryableMemberResult => new QueryResult<object>(
+            new Uri(EntityWithQueryableMemberRequestUri),
+            new object[]
+            {
+                new EntityWithQueryableMember("Test1", CreateQuery<EntityWithSimpleMembers>("entitywithqueryablemembers/Test1/")),
+                new EntityWithQueryableMember("Test2", CreateQuery<EntityWithSimpleMembers>("entitywithqueryablemembers/Test2/"))
+            });
+
+        private static QueryProvider CreateQueryProvider(Uri root)
+        {
+            return new QueryProvider(CreateResourceRetriever(), null, () => new QueryBinder(new EntityValidator()), root);
+        }
+
+        private static IQueryable<T> CreateQuery<T>()
+        {
+            return new Query<T>(CreateQueryProvider(RootUri));
+        }
+
+        private static IQueryable<T> CreateQuery<T>(string subPath)
+        {
+            return new Query<T>(CreateQueryProvider(new Uri(RootUri, subPath)));
+        }
+
+        private static string UniqueIdentifierNameRequestUri =>
+            $"{RootUri}entitywithqueryablemembers?fields=uniqueIdentifier,name";
+
+        private static QueryResult<object> NameResult => new QueryResult<object>(
+            new Uri(UniqueIdentifierNameRequestUri),
+            new object[]
+            {
+                new
+                {
+                    Name = "Test1"
+                },
+                new
+                {
+                    Name = "Test2"
+                }
+            });
+
+        private static QueryResult<object> UniqueIdentifierNameResult => new QueryResult<object>(
+            new Uri(UniqueIdentifierNameRequestUri),
+            new object[]
+            {
+                new
+                {
+                    UniqueIdentifier = "Test1",
+                    Name = "Test1"
+                },
+                new
+                {
+                    UniqueIdentifier = "Test2",
+                    Name = "Test2"
+                }
+            });
+        private static string NameQueryableMemberRequestUri =>
+            $"{RootUri}entitywithqueryablemembers?fields=uniqueIdentifier,name,queryableMember";
+
+        private static QueryResult<object> NameQueryableMemberResult => new QueryResult<object>(
+        new Uri(NameQueryableMemberRequestUri),

```

Comment:
Is indentation correct?

Reply:
Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/1163b4a703b68a8db5e4d2f059e0ab8f52d917b8